### DISCUSSION TITLE
TLS via proxy

### DISCUSSION THREAD

**Discussion by port19x** [Syncplay/syncplay#632] [Q&A] [answered]
I have [syncplay-server](https://github.com/NinetailLabs/syncplay-server) running behind my haproxy.
haproxy terminates TLS for my other services, which are all http based.
Is it possible to do TLS termination at the reverse proxy for syncplay?

In the current setup this does not work

**Comment by port19x** [comment]
https://github.com/Syncplay/syncplay/issues/573

:grimacing:

**Comment by daniel-123** [comment, marked as answer]
As you probably gathered from #573 - Syncplay protocol isn't http at all. It also uses opportunistic TLS.

Both of those make it pretty much completely incompatible with any http proxy. Changing that, at least on the surface, appears like it would involve a lot of work and almost certainly necessitate breaking the compatibility of old clients with new servers (which remains unbroken for almost entire time since Syncplay was first made).
